Allow selecting individual compositions for world builder

gassmoeller opened the pull request:
We have a model setup where using the world builder to prescribe initial compositions is very expensive, however we only want to use it for 1 of our 5 compositions. At the moment it is impossible to call the world builder function only for selected compositions, but this is now added in this PR. 

@MFraters can you take a look if this makes sense?

jdannberg commented on tests/world_builder_select_composition.prm:
```diff
@@ -0,0 +1,72 @@
+end
+
+subsection Postprocess
+  set List of postprocessors = visualization
```
It could be useful to add the composition statistics postptocessor here, which would make it super easy to see that only one composition is nonzero.

jdannberg reviewed (commented):
Nice! I think this will be really useful. Except for the few comments, I think this is great!

MFraters commented:
Yes, this pull request makes sense to me. I agree with you and Juliane that having a default of returning all the compositions is correct. I am on the fence about having the default value being empty or "all", but I am kinda leaning to the side of keeping it empty. There is no reason for the user to provide an empty list, because then they just shouldn't use the world builder for compositions to begin with. And theoretically a user could have a compositional field called "all".

tjhei commented:
2.3 or can this wait?

gassmoeller commented:
I addressed all comments, this should be ready to merge. I chose to use an empty list as default, because it keeps the code simpler. 

This does not need to go into 2.3, it is a significant performance improvement for some models, but only for very specific applications (models using expensive world builder setups with many compositions).

tjhei commented:
I assume you will be running master anyways...

gassmoeller commented:
Yes, for us it does not matter at all.

jdannberg reviewed (approved):
Okay, looks good to me!
